homeassistant/components/shelly/config_flow.py

```diff
@@ -174,12 +190,14 @@ async def async_step_credentials(
                 user_input[CONF_USERNAME] = "admin"
             try:
                 device_info = await validate_input(
-                    self.hass, self.host, self.info, user_input
+                    self.hass, self.host, self.port, self.info, user_input
                 )
             except InvalidAuthError:
                 errors["base"] = "invalid_auth"
             except DeviceConnectionError:
                 errors["base"] = "cannot_connect"
+            except WrongShellyGen:
+                errors["base"] = "wrong_shelly_gen"
```

Comment:
What about gen2+ battery powered devices? You mentioned that they also don't work when configured via a range extender.

Reply:
Fixed with latest aioshelly